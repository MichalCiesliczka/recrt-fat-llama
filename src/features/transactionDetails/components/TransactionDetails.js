import React, { PureComponent, Fragment } from 'react';
// TODO: Add flow
import PropTypes from 'prop-types';
import CircularProgress from '@material-ui/core/CircularProgress';

import TransactionDetailsCard from '../../../components/TransactionDetailsCard/TransactionDetailsCard';

class TransactionDetailsComponent extends PureComponent {
  render() {
    const {
      transaction,
      isDataLoading,
    } = this.props;

    return (
      <Fragment>
        {
          (isDataLoading || !transaction) ? (
            <CircularProgress />
          ) : (
            <TransactionDetailsCard
              transaction={transaction}
            />
          )
        }
      </Fragment>
    );
  }
}

TransactionDetailsComponent.propTypes = {
  // TODO: Fix those types
  transaction: PropTypes.shape({
    borrowerId: PropTypes.number.isRequired,
    creditUsed: PropTypes.number.isRequired,
    currency: PropTypes.string.isRequired,
    fromDate: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    itemId: PropTypes.number.isRequired,
    lenderId: PropTypes.number.isRequired,
    price: PropTypes.number.isRequired,
    promocode: PropTypes.string,
    status: PropTypes.oneOf(['FL_APPROVED', 'PRE_AUTHORIZED_CANCELLED', 'ESCROW', 'PRE_AUTHORIZED', 'PAID', 'CANCELLED', 'QUARANTINED']).isRequired,
    toDate: PropTypes.string.isRequired,
    totalDiscount: PropTypes.number.isRequired,
  }),
  isDataLoading: PropTypes.bool.isRequired,
};

export default TransactionDetailsComponent;
