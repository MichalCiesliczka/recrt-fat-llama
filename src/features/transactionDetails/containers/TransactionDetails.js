import { connect } from 'react-redux';
import { compose } from 'redux';
import { withRouter } from 'react-router';

import {
  selectTransactionById,
  getDataLoadingStatus,
} from '../transactionDetails.selectors';

import TransactionDetailsComponent from '../components/TransactionDetails';

const mapStateToProps = (state, { match: { params } }) => {
  const { id: transactionId } = params;
  return {
    transaction: selectTransactionById(transactionId)(state),
    isDataLoading: getDataLoadingStatus(state),
  };
};

export default compose(
  withRouter,
  connect(mapStateToProps),
)(TransactionDetailsComponent);
