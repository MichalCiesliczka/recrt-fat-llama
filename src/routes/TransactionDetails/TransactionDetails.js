import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import TransactionDetailsContainer from '../../features/transactionDetails';

class TransactionsList extends PureComponent {
  componentDidMount() {
    this.fetchTransactionDetails();
  }

  // TODO: Check if needed
  // componentDidUpdate() {
  //   this.fetchTransactionDetails();
  // }

  fetchTransactionDetails = () => {
    const {
      transactionDetailsRoutine,
      transactionId,
    } = this.props;

    transactionDetailsRoutine(transactionId);
  }

  render() {
    const { transactionId } = this.props;
    return (
      <div>
        <NavLink to="/">
          Get back to list
        </NavLink>
        <p>
          Details for transaction: #
          {transactionId}
        </p>
        <TransactionDetailsContainer />
      </div>
    );
  }
}

TransactionsList.propTypes = {
  transactionDetailsRoutine: PropTypes.func.isRequired,
  transactionId: PropTypes.string.isRequired,
};

export default TransactionsList;
