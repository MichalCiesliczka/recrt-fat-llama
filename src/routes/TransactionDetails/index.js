import Loadable from 'react-loadable';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { withRouter } from 'react-router';

import { transactionDetailsRoutine } from '../../features/transactionDetails/transactionDetails.sagas';

const LoadableComponent = Loadable({
  loader: () => import(/* webpackChunkName: "transactionDetails" */ './TransactionDetails'),
  loading: () => null,
});

const mapStateToProps = (state, { match: { params } }) => {
  const { id: transactionId } = params;

  return {
    transactionId,
  };
};

const mapDispatchToProps = {
  transactionDetailsRoutine,
};

export default compose(
  withRouter,
  connect(mapStateToProps, mapDispatchToProps),
)(LoadableComponent);
