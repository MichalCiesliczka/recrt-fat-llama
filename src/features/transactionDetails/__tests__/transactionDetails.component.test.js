import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import TransactionDetails from '../components/TransactionDetails';

describe('TransactionDetails component', () => {
  const defaultProps = {
    isDataLoading: false,
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TransactionDetails {...defaultProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('display loader with default props', () => {
    const tree = renderer
      .create(<TransactionDetails {...defaultProps} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('display loader when data is loading', () => {
    const dataLoadingProps = {
      isDataLoading: true,
    };

    const tree = renderer
      .create(<TransactionDetails {...dataLoadingProps} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
