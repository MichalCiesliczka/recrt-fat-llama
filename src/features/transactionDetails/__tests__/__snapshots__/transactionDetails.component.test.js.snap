// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionDetails component display loader when data is loading 1`] = `
<div
  className="MuiCircularProgress-root-10 MuiCircularProgress-colorPrimary-13 MuiCircularProgress-indeterminate-12"
  role="progressbar"
  style={
    Object {
      "height": 40,
      "width": 40,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg-15"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle-16 MuiCircularProgress-circleIndeterminate-18"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</div>
`;

exports[`TransactionDetails component display loader with default props 1`] = `
<div
  className="MuiCircularProgress-root-10 MuiCircularProgress-colorPrimary-13 MuiCircularProgress-indeterminate-12"
  role="progressbar"
  style={
    Object {
      "height": 40,
      "width": 40,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg-15"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle-16 MuiCircularProgress-circleIndeterminate-18"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</div>
`;
